### PR TITLE
feat(schedule): layout schedule in desktop and mobile views

### PR DESCRIPTION
**Pull request by flynnhou**
## Types of changes

* **New feature**

## Description

- Realize the schedule design
- Modularize components in the page for better layout and maintenance
- Add mock data in json-server

> Note that this PR doesn't contain event details where they reroute to another page.

## Steps to Test This Pull Request

1. Go to `/conference/schedule` path
2. Play with the page by scrolling, switching day tabs, changing viewports 

## Expected behavior

- Should be compatible with Chrome, Safari, Firefox browsers
- The scroll behaviors should work like last year's

https://user-images.githubusercontent.com/7352613/127785099-bea27dad-d00a-41dd-b948-4d452d5319b3.mp4

## Additional context

Check out the architecture of this page for a higher-level overview: [pycontw-2021 schedule overview](https://docs.google.com/presentation/d/13uW68VyyZq-sTTntyiWf4rNcrDcOn2u3X2rOgbJWS4c/edit?usp=sharing). This is only a temporary documentation. Will update the docs in the repo later.

Feel free to leave comments on the document if it needs to clarify more details.

### CODE REVIEW

**Review by mattwang44 — APPROVED** [review]
Blown away by the PR. LGTM!
Just pushed several commits for some trivial tasks.

**Review by josix — APPROVED** [review]
Looks great to me! Thanks @flynnhou, I got a lot from this PR🙇

**Comment by mattwang44** [comment]
I added another commit for navigating to event pages from the schedule with `LocaleLink`.

File: components/schedule/ScheduleEvent.vue [hunk 1]
```diff
@@ -95,6 +95,22 @@ export default {
...
+            } else if (
+                ['talk', 'tutorial', 'sponsored'].includes(
+                    this.value.event_type,
+                )
+            ) {
+                targetPath = `/conference/${this.value.event_type}/${this.value.event_id}/`
+            } else {
+                return ''
+            }
+            return targetPath
```
Comment (flynnhou) [inline components/schedule/ScheduleEvent.vue:112]:
```suggestion
const { event_type: eventType, event_id: eventId } = this.value
if (eventType === 'keynote') {
    return '/conference/keynotes/'
} else if (['talk', 'tutorial', 'sponsored'].includes(eventType)) {
    return `/conference/${eventType}/${eventType}/`
} else {
    return ''
}
```

**Pull request merged**